io/src/main/scala/fs2/io/file/Files.scala

```diff
@@ -0,0 +1,510 @@
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package fs2
+package io
+package file
+
+import scala.concurrent.duration._
+
+import cats.effect.kernel.{Async, Resource, Sync, Temporal}
+import cats.syntax.all._
+
+import java.io.IOException
+import java.nio.channels.{FileChannel, FileLock}
+import java.nio.file.{Files => JFiles, _}
+import java.nio.file.attribute.{BasicFileAttributes, FileAttribute, PosixFilePermission}
+import java.util.stream.{Stream => JStream}
+
+import fs2.io.CollectionCompat._
+
+trait BaseFiles[F[_]] {
+
+  /**
+    * Copies a file from the source to the target path,
+    *
+    * By default, the copy fails if the target file already exists or is a symbolic link.
+    */
+  def copy(source: Path, target: Path, flags: Seq[CopyOption] = Seq.empty): F[Path]
+
+  /**
+    * Creates a new directory at the given path.
+    */
+  def createDirectory(path: Path, flags: Seq[FileAttribute[_]] = Seq.empty): F[Path]
+
+  /**
+    * Creates a new directory at the given path and creates all nonexistent parent directories beforehand.
+    */
+  def createDirectories(path: Path, flags: Seq[FileAttribute[_]] = Seq.empty): F[Path]
+
+  /**
+    * Deletes a file.
+    *
+    * If the file is a directory then the directory must be empty for this action to succeed.
+    * This action will fail if the path doesn't exist.
+    */
+  def delete(path: Path): F[Unit]
+
+  /**
+    * Like `delete`, but will not fail when the path doesn't exist.
+    */
+  def deleteIfExists(path: Path): F[Boolean]
+
+  /**
+    * Recursively delete a directory
+    */
+  def deleteDirectoryRecursively(path: Path, options: Set[FileVisitOption] = Set.empty): F[Unit]
+
+  /**
+    * Creates a stream of [[Path]]s inside a directory.
+    */
+  def directoryStream(path: Path): Stream[F, Path]
+
+  /**
+    * Creates a stream of [[Path]]s inside a directory, filtering the results by the given predicate.
+    */
+  def directoryStream(path: Path, filter: Path => Boolean): Stream[F, Path]
+
+  /**
+    * Creates a stream of [[Path]]s inside a directory which match the given glob.
+    */
+  def directoryStream(path: Path, glob: String): Stream[F, Path]
+
+  /**
+    * Checks if a file exists.
+    *
+    * Note that the result of this method is immediately outdated. If this
+    * method indicates the file exists then there is no guarantee that a
+    * subsequence access will succeed. Care should be taken when using this
+    * method in security sensitive applications.
+    */
+  def exists(path: Path, flags: Seq[LinkOption] = Seq.empty): F[Boolean]
+
+  /**
+    * Moves (or renames) a file from the source to the target path.
```

Comment:
Is this guaranteed to be atomic?

Reply:
It is if you pass `StandardCopyOption.ATOMIC_MOVE`

Reply:
Might be worth adding that to the docs - I wouldn't have know how to find that necessarily

Reply:
Yeah - want to open an issue for these types of doc improvements? Note this PR didn't introduce these methods but rather just moved them from `fs2.io.file` to `Files`.